ingest/src/main/java/com/microsoft/azure/kusto/ingest/AzureStorageHelper.java

```diff
@@ -99,8 +149,36 @@ private void copyStream(InputStream inputStream, OutputStream outputStream, int
     }
 
     String getBlobPathWithSas(CloudBlockBlob blob) {
+        validateIsNotNull(blob, "blob is null");
+
         StorageCredentialsSharedAccessSignature signature = (StorageCredentialsSharedAccessSignature) blob.getServiceClient().getCredentials();
         return blob.getStorageUri().getPrimaryUri().toString() + "?" + signature.getToken();
     }
 
+    long getBlobSize(String blobPath) throws StorageException, URISyntaxException {
+        validateIsNotEmpty(blobPath, "blobPath is empty");
+
+        CloudBlockBlob blockBlob = new CloudBlockBlob(new URI(blobPath));
+        blockBlob.downloadAttributes();
+        return blockBlob.getProperties().getLength();
+    }
+
+    // Validation functions:
+    private void validateIsNotEmpty(String str, String message) {
+        if(StringUtils.isEmpty(str)){
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    private void validateIsNotNull(Object obj, String message) {
```

Comment:
The usage of Objects.requireNonNull() is in my opinion less suitable to argument validations on methods, because it throws NullPointerException and not IllegalArgumentException. It is maybe better for using in constructors.

Anyway, the implementation of my method is exactly the same, except the exception it throws.

Reply:
I think there's a debit what more appropriate and I didn't find a clear answer. 
I don't see the difference between a method and a constructor - both get parameters and need to validate them...
I like to use builtin things rather then develop my own, also the Objects class looks cleaner and that's why I choose to use it in the first place.

Reply:
Changing it will make this validation function less compatible with other validation functions in this class, and it will throw a NullPointerException.
Anyway, if you think it should be changed, I will change it.

Reply:
You can convert the NullPointerException to IllegalArgumentException is needed